### ISSUE TITLE
[BUG] Search: IndexDocumentsAsync not serializing complex types

### ISSUE THREAD

**Issue by kirk-marple** [Azure/azure-sdk-for-net#17685]
**Describe the bug**
Please provide the description of issue you're seeing.

Not getting complex fields serialized properly into Cognitive Search index.  Tried with latest beta of search SDK (11.2.0-beta.2).

Appears to be similar to this bug [https://github.com/Azure/azure-sdk-for-net/issues/10944](10944)

**Expected behavior**
What is the expected behavior?

Prior to upgrading to v11 of API, I would get valid complex types serialized into my index.  I expected this to not change with v11.

**Actual behavior (include Exception or Stack Trace)**

In Search Explorer, my complex type is serialized as null:

![image](https://user-images.githubusercontent.com/13594550/102931985-50e41a80-4454-11eb-81b8-add7441084af.png)

![image](https://user-images.githubusercontent.com/13594550/102931874-0f536f80-4454-11eb-9636-d2cc1f7fe42e.png)

```        
[JsonPropertyName("site")]
public SiteIndex Site
{
    get; set;
}

    public class SiteIndex
    {
        [SimpleField(IsFilterable = true, IsFacetable = true)]
        [JsonPropertyName("siteLocation")]
        public string SiteLocation
        {
            get; set;
        }

        [SimpleField(IsFilterable = true, IsFacetable = true)]
        [JsonPropertyName("siteType")]
        public string SiteType
        {
            get; set;
        }

        [SimpleField(IsFilterable = true, IsFacetable = true)]
        [JsonPropertyName("fileConnectorType")]
        public string FileConnectorType
        {
            get; set;
        }

        [SimpleField(IsFilterable = true, IsFacetable = true)]
        [JsonPropertyName("region")]
        public string Region
        {
            get; set;
        }
    }
```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

`                    
return await client.IndexDocumentsAsync(IndexDocumentsBatch.MergeOrUpload(dd));
`

**Environment:**
 - Name and version of the Library package used: [e.g. Azure.Storage.Blobs 12.2.0] 

Azure.Search.Documents 11.2.0-beta.2

 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): [e.g. Azure AppService or Windows 10 .NET Framework 4.8]

.NET 5.0, xUnit test runner
Windows 10

 - IDE and version : [e.g. Visual Studio 16.3]

Visual Studio 16.9.0 Preview 2

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.  Please be aware that due to the holidays, responses may be delayed.

**Comment by kirk-marple** [comment]
@tg-msft I've actually figured this out.  It was an issue with the lack of polymorphic serialization in System.Text.Json.

https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to?pivots=dotnet-5-0#polymorphic-serialization

I was serializing a list of "EntityIndexBase" references, and it wasn't properly serializing the properties from the parent class "KnowledgeEntityIndex".  I changed this to pass down the proper type as a generic type, and it now is working.

Also, it seemed like I needed to set this DefaultIgnoreCondition to properly handle an array of sub-index objects:

                Serializer = new JsonObjectSerializer(new JsonSerializerOptions
                {
                    DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
                    PropertyNamingPolicy = JsonNamingPolicy.CamelCase
                })

Closing this issue.
